### ISSUE TITLE
Widget does not update until clicked

### ISSUE THREAD

**Issue by Winell** [HanSolo/Gluco-Tracker#13]
On my iPhone 13 pro and my wife’s iPhone 13 the widget does not automatically update the value until pressed. Application has access enabled to run in the background in IOS settings.

**Comment by HanSolo** [comment]
Yep that problem is related to the fact that on iOS, the operating systems decides when to update the widget. So usually it will only get updated once every 20 min. There is a way around that but that would involve an external server that can send push notifications to update widgets but that is out of my possibilities at the moment. So unfortunately updating the widget is not under my control but I will again look into it because I also would like to have that.

**Comment by Winell** [comment]
Thank you. I read in the closed issues that the refresh rate is triggered by an algorithm in IOS.  Any way for the end user to have it trigger more often, ie interact with the widget more or less often?

**Comment by HanSolo** [comment]
Yep you can get up to 1 update every 5 minutes as far as I know and that will be calculated by the operating system on how often you use the widget. To be honest I did not saw a big decrease in update times when I've tried to use it more often but who knows what Apples AI is doing there.
